python/revng/profile/base.py

```diff
@@ -0,0 +1,13 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+from abc import ABC, abstractmethod
+
+from revng.scripting import Project
+
+
+class Profile(ABC):
+    @abstractmethod
+    def get_project(self) -> Project:
```

Comment:
Not sure if a common interface for this class is useful, I'd drop it for the time being